### ISSUE TITLE
[BUG] Re2 Requires a higher NODE_MODULE_VERSION

### ISSUE THREAD

**Issue by ananzh** [opensearch-project/OpenSearch-Dashboards#1436]
**Describe the bug**

node-re2 so that we can safely use the regular expressions specified by end-users in Timelion
see version conflict error when using v1.2 on arm instance.

v1.2:
<img width="1533" alt="Screen Shot 2022-04-07 at 22 52 08" src="https://user-images.githubusercontent.com/79961084/162476818-6a07f2cd-db15-4c44-8ed3-2d9eb5cf7788.png">

main (2.0):
<img width="1654" alt="Screen Shot 2022-04-08 at 10 13 33" src="https://user-images.githubusercontent.com/79961084/162490263-6f8244b0-abae-464a-8745-f8cdc1047b58.png">

**To Reproduce**
create v1.2 graviton2 domain                                                                                             
logon to OpenSearch and load sample web log data                                                                   
go to visualize and select to create a timelion visualization       
run expression:
```                                      
.es(*).label('all', '.*')
```

**Expected behavior**
fix the error

**OpenSearch Version**
v1.2 +

**Dashboards Version**
v1.2 +

**Plugins**
no plugin

**Host/Environment (please complete the following information):**
 - Browser and version [e.g. 22]

**Comment by ananzh** [comment]
Issue is only limit to arm. don't see issue on other platform:
<img width="1656" alt="Screen Shot 2022-04-08 at 11 57 30" src="https://user-images.githubusercontent.com/79961084/162506665-be6dfc54-bd82-4963-b5aa-a6c5ed72d29e.png">

**Comment by ananzh** [comment]
My understanding from these tests:
Seems there are two environments with different Node versions. One environment is used to build re2 on arm, another is to run it which is the version when run `node --v`.

When we build arm, we download from: https://d1v1sj258etie.cloudfront.net/node-re2/1.15.4/linux-arm64-64.gz
then we extract it to the target path. 

```
 info [  opensearch-dashboards  ] Patching platform-specific native modules
   │ debg Patching re2 binaries from https://d1v1sj258etie.cloudfront.net/node-re2/1.15.4/linux-arm64-64.gz to /home/anan/work/OpenSearch-Dashboards/build/opensearch-dashboards-2.0.0-SNAPSHOT-linux-arm64/node_modules/re2/build/Release/re2.node
platformName: linux-arm64
archive: {
  url: 'https://d1v1sj258etie.cloudfront.net/node-re2/1.15.4/linux-arm64-64.gz',
  sha256: '19fa97f39d4965276382429bcd932dd696915f711663e7cef9b0a304b3e8e6f7'
}
archiveName: linux-arm64-64.gz
downloadPath: /home/anan/work/OpenSearch-Dashboards/.native_modules/re2/linux-arm64-64.gz
extractPath: /home/anan/work/OpenSearch-Dashboards/build/opensearch-dashboards-2.0.0-SNAPSHOT-linux-arm64/node_modules/re2/build/Release/re2.node
   │ debg Deleting patterns: [
   │        '/home/anan/work/OpenSearch-Dashboards/build/opensearch-dashboards-2.0.0-SNAPSHOT-linux-arm64/node_modules/re2/build/Release/re2.node'
   │      ]
   │ debg Deleted 1 files/directories
   │ debg Attempting download of https://d1v1sj258etie.cloudfront.net/node-re2/1.15.4/linux-arm64-64.gz 19fa97f39d4965276382429bcd932dd696915f711663e7cef9b0a304b3e8e6f7
   │ debg Downloaded https://d1v1sj258etie.cloudfront.net/node-re2/1.15.4/linux-arm64-64.gz and verified checksum
   │ succ ✓ 0 sec
```

**Comment by seraphjiang** [comment]
Hi team @opensearch-project/opensearch-dashboards-core  
could someone help me to understand what's current build process and environment for ARM artifacts?

**Comment by AMoo-Miki** [comment]
@seraphjiang, I maintain the arm64 builds for node-re2 in a personal capacity. It involves a manual triggering of `npm i re2@1.15.4` on an arm64 environment. Then the artifacts are uploaded for consumption.

@ananzh can you please verify that these work for the 1.x branch?
```
https://d1v1sj258etie.cloudfront.net/node-re2-experimental/1.15.4/linux-arm64-64.gz
https://d1v1sj258etie.cloudfront.net/node-re2-experimental/1.15.4/sha256sum.txt.asc
```

**Comment by ananzh** [comment]
v1.2 result after updates:
<img width="1575" alt="Screen Shot 2022-04-08 at 14 21 29" src="https://user-images.githubusercontent.com/79961084/162533842-53c3f933-a992-4316-a4da-6d9e12960c21.png">

**Comment by ananzh** [comment]
v1.3 result after fix:
OS 1.3
OSD 1.3.1

<img width="1570" alt="Screen Shot 2022-04-08 at 14 32 41" src="https://user-images.githubusercontent.com/79961084/162535073-03c09359-ab2b-44f2-8fb3-04df3157e226.png">

**Comment by seraphjiang** [comment]
> node-re2-experimental

Thanks @AMoo-Miki for providing node-re2 redistribution. my original question is what's our build process for OSD arm artifacts. Now, I have one more question about build/redistribution process for node-re2 ARM

There are opportunities to improve here, shall we discuss, at least add an item to backlog.

**Comment by AMoo-Miki** [comment]
There are the final releases. They are `.tar.gz` files and as such, `src/dev/build/tasks/patch_native_modules_task.ts` needs to be modified.

@ananzh, please test them as I don't have the means to do so and let me know if you face any problems.

### re2@1.15.4

Node v10: https://d1v1sj258etie.cloudfront.net/node-re2/releases/download/1.15.4/linux-arm64-64.tar.gz
Node v14: https://d1v1sj258etie.cloudfront.net/node-re2/releases/download/1.15.4/linux-arm64-83.tar.gz
https://d1v1sj258etie.cloudfront.net/node-re2/releases/download/1.15.4/sha256sum.txt.asc

### re2@1.17.4

Node v10: https://d1v1sj258etie.cloudfront.net/node-re2/releases/download/1.17.4/linux-arm64-64.tar.gz
Node v14: https://d1v1sj258etie.cloudfront.net/node-re2/releases/download/1.17.4/linux-arm64-83.tar.gz
https://d1v1sj258etie.cloudfront.net/node-re2/releases/download/1.17.4/sha256sum.txt.asc

**Comment by kavilla** [comment]
Existing code seeing build failures: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1446

**Comment by seraphjiang** [comment]
We should back port to 1.3 as promised to customer

**Comment by ananzh** [comment]
@AMoo-Miki 
<img width="1580" alt="Screen Shot 2022-04-13 at 09 15 53" src="https://user-images.githubusercontent.com/79961084/163225203-01aceb09-e2fc-4a12-a847-b9e024b2a331.png">

Error looks to me that `./build/Release/re2` is not found. However, `./build/Release/re2` is there in node_modules/re2 . Any other reasons you could think about regarding this error?

**Comment by seraphjiang** [comment]
> [BUG]re2 regex lib not working in ARM #1361

re-open the issue as we didn't get answer about backport, but it is critical.
I knew there is no planned 1.3.2 yet, could we at least provide estimation, what takes to back port to 1.x/1.3.x

**Comment by tmarkley** [comment]
The `backport x.x` labels are for PRs. We should be using the `vx.x.x` labels to indicate the targeted releases.

**Comment by tmarkley** [comment]
@seraphjiang this will be released with v1.3.2: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/release-notes/opensearch-dashboards.release-notes-1.3.2.md
